### PR TITLE
Add profile for testsuite

### PR DESCRIPTION
**Pull request by Bischoff**
Add profile for testsuite, in Testsuite/ top-level directory.

### CODE REVIEW

**Comment by aaannz** [comment]
@joesusecom @abhinav-puri as PM what do you think about this?
As this is customer facing github repository, should it also contain testsuite data with links to SUSE internal repositories?
Other idea may be not to merge it to master but have i.e. devel/test branch where would be the same profiles at the same paths but with SUSE SLE repositories needed for successful build and useless to public.

PR itself is good and validated by a successful testsuite run.

**Comment by aaannz** [comment]
By internal repositories I mean for packages - e.g. `http://euklid.nue.suse.com/mirror/SuSE/build.suse.de/SUSE/Products/SLE-SERVER/12-SP3/x86_64/product/` and other.

**Comment by Bischoff** [comment]
> By internal repositories I mean for packages - e.g. 
> http://euklid.nue.suse.com/mirror/SuSE/build.suse.de/SUSE/Products/SLE-SERVER/12-SP3/x86_64/product/ and other.

We consider that external people running the testsuite need have access to SLE products. This will be the case in other part of the testsuite too. Is there a problem with that?

Or we could use openSUSE Leap.

**Comment by Bischoff** [comment]
I had not the time to comment on the other half of Ondrej's message

> Other idea may be not to merge it to master but have i.e. devel/test branch where would be the same
> profiles at the same paths but with SUSE SLE repositories needed for successful build and useless to
> public.

I like the idea. We could call the new branch "testsuite" ;-).

Well, I would prefer in "master", but this would be an reasonable solution.


> PR itself is good and validated by a successful testsuite run.

Thank you Ondrej for the nice words.

**Comment by Bischoff** [comment]
> at least I will explain at the README what Testsuite is (and IMHO we should explain it for the other
> folders as well).

Already done, look at the `README.md` in this PR.

File: README.md [hunk 1]
```diff
@@ -3,3 +3,5 @@
 This repository contains sources that can be used in SUSE Manager image profiles to build various images.
 
 Repository is organized by build types and then individual image names.
+
+The special "Testsuites" directory contains profiles used by SUSE Manager cucumber testsuite.
```
Comment (juliogonzalez) [inline README.md]: I'd create a list, explaining what each of the directories has:

```
Repository is organized by build types and then individual image names.

Build types, as directories at the root folder:

- `Containers`: Whatever
- `OSimages`: Whatever
- `Testsuites`: Profiles used by SUSE Manager cucumber testsuite.
```

Comment (Bischoff) [reply README.md]: ok

**Comment by Bischoff** [comment]
Done (better README)

**Comment by mcalmer** [comment]
AFAIK the Retail people needs to decide. If they are ok, we can do it.

**Comment by aaannz** [comment]
This is not only about Retail. This is about image templates basic sources and examples for customer as a whole. By basic source I mean something that can build an image as is and also customer can use to create their own custom image templates.
So I guess that toplevel directory separation for Testsuite is fine, but because this being in documentation in a way "go there and pick something" I was quite hesitant. And that's why I also added @joesusecom and @abhinav-puri
Quote from [documentation](https://github.com/SUSE/doc-susemanager/blob/develop/adoc/advanced_topics_image_management.adoc#example-of-kiwi-sources):
> SUSE provides examples of fully functional image sources at the SUSE/manager-build-profiles public GitHub repository.

E.g. when @cbosdo finishes virtualization support they may want their own KVM or XEN templates here as a basic sources for customers.

**Comment by Bischoff** [comment]
@aannz If it still itches you to have this top-level directory in same branch (and apparently it does), I'm okay with your idea of a separate `testsuite` branch.

Please decide.

**Comment by jsrain** [comment]
@Bischoff @aaannz I can live with having internal URLs disclosed in the test suite as long as it is obvious enough that if anyone wants to run the test suite, he needs to change it. Not that I'm entirely happy of no separation, but we should not overcomplicate our lives.

**Comment by Bischoff** [comment]
@jsrain for the disclosure of internal URLs, as @juliogonzalez  said it was asked to legal.

Since Uyuni itself will have public URLs, we should be able to switch relatively soon.

If not, we could use openSUSE URLs.

Please notice it's more general that this PR only, it goes beyond containers and os images, it also concerns the server, the proxy and all the clients we test. Ping @moio for the consequences on sumaform.

**Comment by moio** [comment]
https://github.com/SUSE/spacewalk-testsuite-base has been on public GitHub since inception despite testing SUSE Manager only. There is no problem with that.

Internal infrastructure URLs are also not a problem.

I agree with the general view that we should keep our lives as simple as possible.

Thus, I expect testsuites to accompany their respective product versions their branches. So SUSE/spacewalk/Manager-3.0 should contain the 3.0 testsuite, SUSE/spacewalk/Manager-3.1 the 3.1 testsuite and SUSE/spacewalk/Manager-3.2 the 3.2 testsuite which includes Retail. uyuni-project/uyuni/master should contain the Uyuni testsuite, which is what we called the Manager testsuite until yesterday (next major version).

At this point, it does not really make sense to work on a Retail testsuite in uyuni-project/uyuni/master, so all work should happen in SUSE/spacewalk/Manager-3.2 unless I am missing something.

When work on a future Retail version based on the next major Manager version will start, there will be no other chance but to start working on uyuni-project/uyuni master. At that point all work on the 3.2 testsuite will have to be forward-ported, made public, and made optional (default disabled) to be nice with the community.

**Comment by Bischoff** [comment]
Thanks @moio for the idea.

I see both advantages and a drawback to this suggestion:
* advantage: when we have to modify something related to profiles, we can do it both in the testsuite and in the profile in one single pull request
* other advantage: it does not mix examples with testsuite stuff, which was @aaannz main concern
* drawback: it will be a bit less realistic a test than if we tested from here (manager-build-profiles)

I think the advantages overcome the drawback. Would everyone agree with Silvio's solution? If yes, I close this PR and create the profiles in the testsuite itself.

**Comment by mbologna** [comment]
I agree with having the testsuite profiles not in this PR but rather in the testsuite itself.

**Comment by Bischoff** [comment]
OK, creating a card to do it (under waterline) on next sprint. Closing here.

**Pull request closed without merging**